pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -197,6 +197,7 @@ private PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopG
         } catch (Throwable t) {
             shutdown();
             shutdownEventLoopGroup(eventLoopGroup);
+            closeConnectionPool(cnxPool);
```

Comment:
The changes on `PulsarClientImpl` are not related with gson -> jackson and should go on their own separate commit.

Reply:
👍